tailcfg/tailcfg.go

```diff
@@ -875,6 +876,7 @@ type Hostinfo struct {
 	UserspaceRouter opt.Bool       `json:",omitempty"` // if the client's subnet router is running in userspace (netstack) mode
 	AppConnector    opt.Bool       `json:",omitempty"` // if the client is running the app-connector service
 	ServicesHash    string         `json:",omitempty"` // opaque hash of the most recent list of tailnet services, change in hash indicates config should be fetched via c2n
+	ExitNodeID      StableNodeID   `json:",omitzero"`  // the client’s selected exit node, empty when unselected.
```

Comment:
:question: Is Hostinfo the best place to send this information back to Control? If we don’t send it back through Hostinfo, how shall we handle MapUpdateDiff?

Reply:
The short answer is that yes, it is the best place. The long answer is that it really shouldn’t be the best place, but there’s no better place to kick off updates of the network map.